src/librbd/cache/ParentCacheObjectDispatch.cc

```diff
@@ -130,6 +141,68 @@ void ParentCacheObjectDispatch<I>::handle_read_cache(
   on_dispatched->complete(r);
 }
 
+template <typename I>
+void ParentCacheObjectDispatch<I>::read_parent(
```

Comment:
I'd hate to have to re-implement this logic. With the in-progress crypto layer, it will also have an issue w/ handling reading from parents so I think we will eventually need a longer term solution. I wonder if for now it would just be easier to return a dispatch result of `CONTINUE` so that it moves to the next layer for now. Otherwise, I think it would be helpful to have some API or utility function to wrap all this read-from-parent logic up so that it's not duplicated.

Reply:
Actually, just returning `CONTINUE` dispatch result was my initial implementation, but it looked not very nice to me -- more potential rados calls, which would make the parent cache less effective. 

I am rather for your second suggestion, i.e. I will try to dedup.